### ISSUE TITLE
Implement ensure process ready

### ISSUE THREAD

**Issue by iLLiCiTiT** [ynput/ayon-kitsu#71]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Please describe the feature you have in mind and explain what the current shortcomings are?

Implement functionality that makes sure everything is set up for certain processes.

At this moment all the preparation happens during Tray initialization and startup, e.g. login. But when webactions will be used, there is no option to run or validate that preparations, because tray is not running, or is running in different process.

Webactions, especially those who require publishing, will ask ayon-core to ask all addons to prepare and validate their requirements. That part has to be implemented for shotgrid.

### How would you imagine the implementation of the feature?

Implementation definition in ayon-core is here https://github.com/ynput/ayon-core/pull/781 . Addon has to override `ensure_is_process_ready` where should happen whatever it needs to do. The method will receive `ProcessContext` where are metadata information about the started project, e.g. project name (does not have to be filled - e.g. traypublisher), if is in headless mode, which addon and version triggered the preparation.

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the enhancement request.

**Comment by MustafaJafar** [comment]
Please correct me.

So, getting and setting credentials should be implemented in `KitsuAddon.ensure_is_process_ready` as well as `KitsuAddon.tray_start`? 
and we can take reference from ayon ftrack 
https://github.com/ynput/ayon-ftrack/blob/a9b5314e308c5d378903d7eb7aaee451e01c6f5e/client/ayon_ftrack/ftrack_addon.py#L194-L241

**Comment by iLLiCiTiT** [comment]
> So, getting and setting credentials should be implemented in KitsuAddon.ensure_is_process_ready as well as KitsuAddon.tray_start?

Yes, but `ensure_is_process_ready` should probably be more strict.

**Comment by MustafaJafar** [comment]
Tagging @EmberLightVFX

**Comment by EmberLightVFX** [comment]
Don't see how it could be a bad thing implementing this 😺
